traitsui/tests/editors/test_instance_editor.py

```diff
@@ -67,16 +97,55 @@ def test_simple_editor(self):
     def test_custom_editor(self):
         obj = ObjectWithInstance()
         tester = UITester()
-        with tester.create_ui(obj, dict(view=get_view("custom"))) as ui:
+        with tester.create_ui(obj, {'view': custom_view}) as ui:
             value_txt = tester.find_by_name(ui, "inst").find_by_name("value")
             value_txt.perform(KeySequence("abc"))
             self.assertEqual(obj.inst.value, "abc")
 
+    def test_custom_editor_with_selection(self):
+        obj = ObjectWithList()
+        tester = UITester()
+        with tester.create_ui(obj, {'view': selection_view}) as ui:
+            # test that the current object is None
+            self.assertIsNone(obj.inst)
```

Comment:
Im surprised by this, because `inst` is defined as `Instance(NamedInstance, ())` which I thought should have a default value of `NamedInstance()`.

Reply:
I believe this is because we have the view:
```
selection_view = View(
    Item(
        "inst",
        editor=InstanceEditor(name='inst_list'),
        style='custom',
    ),
    buttons=["OK"],
)
```
so the value of `inst` is matching the fact that we have not yet selected an instance from the list `inst_list`.

Reply:
a call to configure_traits in the test visually confirmed this for me.

I do still find this a little confusing looking though as my instinct says `Instance(NamedInstance, ())` defaults to `NamedInstance()` and `Instance(NamedInstance)` defaults to `None`.